### ISSUE TITLE
Setting `macos_window_background_blur` above 0 leads to high GPU usage

### ISSUE THREAD

**Issue by BhawickJain** [wez/wezterm#5555]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20240609-095134-f5e496eb

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When setting `macos_window_background_blur` to any value above 0, it leads to my macbook to heat up and spin its fans up. The Activity App reports GPU usage of 20-40%. I have an integrated GPU.

### To Reproduce

MacBook Details:
MacBook Pro (Retina, 15-inch, Mid 2014)
2.2 GHz Quad-Core Intel Core i7
16 GB 1600 MHz DDR3
Intel Iris Pro 1536 MB (Integrated Graphics)

### Configuration

```lua
local wezterm = require 'wezterm'
local config = {}


config.font_size                     = 16
config.font                          = wezterm.font('JetBrains Mono', { weight = 'Bold', italic = true })
config.hide_tab_bar_if_only_one_tab  = true
config.hide_mouse_cursor_when_typing = true
config.default_cursor_style          = 'SteadyBlock'
config.window_decorations            = "RESIZE"
config.window_background_opacity     = 0.80
config.macos_window_background_blur  = 30

-- old laptop friendy
config.animation_fps = 1
config.front_end = "OpenGL"


return config

```

### Expected Behavior

I expect GPU usage to be relatively low perhaps under 10% when having `macos_window_background_blur` set to above 0. I also don't expect Window Server GPU to idle around 20%.

I have tried not setting `animation_fps` and `front_end` with no substantial difference in GPU usage.

### Logs

```
Debug Overlay
wezterm version: 20240609-095134-f5e496eb x86_64-apple-darwin
Window Environment: macOS 11.7.10 (20G1427)
Lua Version: Lua 5.4
OpenGL: Intel Iris Pro OpenGL Engine 4.1 INTEL-16.5.14
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
>
```

### Anything else?

When I set `macos_window_background_blur ` to 0, the Window Server GPU usage drops to single figures. `window_background_opacity` does not make any difference on GPU load at any value. I am switching from alacritty which has a boolean Blur option which does not cause the elevated GPU usage.

**Comment by martin-braun** [comment]
I searched for a very long time for the culprit of my WindowServer being at 30% most of the times, and since wezterm is my entry application that always runs, this seems to be part of my problem. I, however, cannot confirm any single figure digits consistently when setting `window_background_opacity` to 0, but it significantly improved.

We want to have a good looking OS and a good looking terminal though. For lightly tasks while being plugged in, I wanted to have blur on 20 and also some opacity on the background, but I linked the accessibility "Increased Contrast" to my menu bar, which will also enable "Reduced Transparency". Alternatively, "Reduced Transparency" can be toggled as well.

If `reducedTransparency" is toggled, it seem to trigger `wezterm` to reload its config. I don't know why, but this is golden behavior, because now I can just read the setting in my configuration and act accordingly:

```lua
local isMac = wezterm.target_triple == 'x86_64-apple-darwin' or wezterm.target_triple == 'aarch64-apple-darwin'

T.window_background_opacity = 0.7
if isMac then
	local hndl = io.popen("defaults read com.apple.universalaccess reduceTransparency")
	local reduceTransparency = hndl:read("*a"):sub(1, 1)
	hndl:close()
	T.window_background_opacity = reduceTransparency == "1" and 1 or T.window_background_opacity
	T.macos_window_background_blur = reduceTransparency == "1" and 20 or 0
end
```

Now my work around is, whenever I'm on the go, I simply increase my contrast or reduce the transparency. Sitting in the sun with my MacBook would cause it to quickly overheat and lose a lot of battery, because of the WindowServer eating so much GPU. This is my work-around. Even if this issue will be fixed, it might be a good option in general. Increased contrast is also great when the sun light impairs vision.

**Comment by BhawickJain** [comment]
That's a lovely workaround. I generally like using blur when extending the terminal window over something noisy while I work. 

I have been using a bash script to quickly adjust the blur level via an alias.
```bash
#!/bin/bash

set -eu

BLUR=${1:-00} # no blur as default value
wezterm_path="$HOME/.config/wezterm/wezterm.lua"

if [[ ! (-f "$wezterm_path") ]]; then
    echo "error file not found: $wezterm_path"
    exit 1
fi
    
# input blur input value must be `00-99`
if [[ "$BLUR" =~ ^[0-9]{2}$ ]]; then
    perl -pi -e "s/(?<=config.macos_window_background_blur  = )[0-9]+$/$BLUR/g" $wezterm_path
else
    echo "invalid blur value: $BLUR"
    exit 1
fi
```
It's quick and dirty, substituting the blur value in the config---wezterm hot reloads just fine. I can enter any double digit value `00-99`, with no blur as the default value.

Ideally I would call Wezterm's CLI to change the runtime blur temporarily, which alacritty's CLI provides. As far as I am aware, this is not possible on Wezterm. 

Either way having the option to adjust the blur on the fly has been very useful.

**Comment by martin-braun** [comment]
@BhawickJain That's also a possibility, although I would always try to refrain from monkey patching as my configurations are part of a git repository, but there is nothing wrong to it using this as a work around 👍

If I would solve it the way you need it, I would rather store the blur value in a different file and then use the WezTerm CLI to trigger `reload_configuration`. In the configuration itself you could read the new value from the new file and set it accordingly (with a fallback value if the file is not found). It would avoid the monkey patching altogether.

I hope the slowness of the blur effect can be worked on to improve it over time.

**Comment by BhawickJain** [comment]
@martin-braun Yes totally! These are good options to avoid the monkey patching. I think with `wezterm.add_to_config_reload_watch_list(path)` you can explicitly tell WezTerm to monitor additional files, so there is a neat scope of possible workarounds.

**Comment by wez** [comment]
FWIW, all wezterm does to enable blur is call a couple of macOS functions to request that the system enable and apply the blur. I'm not sure that there is anything that wezterm can do to make that effect cheaper.

It's also possible that your older intel mac is not as efficient at this kind of operation as an M1 mac.

You might also consider setting `max_fps` to a smaller value to limit the upper bound on FPS when using this option.
